src/main/java/com/epam/eighty/service/TopicService.java

```diff
@@ -16,4 +17,6 @@
     void updateTopic(Topic topic);
     void deleteTopic(Long id);
     Topic createTopic(Topic topic, Long id);
+    Long getIdOfLastNotDeletedTopic(List<Long> topicIds);
+    Topic getTopicWithChildsTillTopicWithId(Long id);
```

Comment:
Maybe getTopicWithChildrenToCurrentTopic or getRootWithSubtopicsToCurrentTopic but I'm not sure that they are suitable they are too complex and it is not clear what id param means. What are your variants?